server/block_processor.py

```diff
@@ -540,6 +540,12 @@ def advance_txs(self, txs):
             # Spend the inputs
             if not tx.is_coinbase:
                 for txin in tx.inputs:
+                    # NOTE(gabriel-samfira): Be forgiving if someone wants
+                    # to rebase and has not implemented is_spendable in their
+                    # custom TxInput
+                    check_spendable = getattr(txin, 'is_spendable', None)
+                    if check_spendable and check_spendable() is False:
+                        continue
```

Comment:
I'm sorry but having something coin-specific in these 2 core loops is not acceptable for performance reasons

Reply:
Okay, fair enough. Will look for alternatives. I must note, that this will probably happen with any coin that has stealth addresses, not just this one. So it may benefit other coins as well. Any suggestion as to where this would be better placed?

Thanks for the quick review :D.

Reply:
The problem with your approach is a) the test is inefficient, but more b) it is done per tx_in.

I would prefer some kind of pre- or post-processing loop that did whatever was necessary.

Also, I think you will need to do something for block reorgs, because that requires undoing what was done when processing a block, so tends to incorporate similar logic.  I don't see that in your code.

I would like to require tests for new functionality such as this, but as I don't (yet) have them for the standard code paths I feel it's hypocritical to ask others to do that; realistic tests for block processing will require a fair amount of work.

Reply:
I may have mis-spoken and you it seems you did cover the reorg case.

Reply:
Indeed. These changes were needed to have electrumx work with stealth transactions. Inputs originating from stealth inputs have [a different format](http://paste.ubuntu.com/p/jzYmYZD9kk/), and the ```scriptSig``` field includes [OP_RETURN](https://en.bitcoin.it/wiki/OP_RETURN).

Will think about a more efficient way to do this, but it is great to get input from you :). Apologies for not including tests as well.

Reply:
Apropos, a similar check to what I added, seems to happen in:

https://github.com/kyuupichan/electrumx/blob/master/server/block_processor.py#L545

```hashX_from_script()``` [seems to do the same kind of check](https://github.com/kyuupichan/electrumx/blob/master/lib/coins.py#L131-L137) and it is done for every output. The check is just done in another function.

I will still think about finding a better solution for my change though. Just thought it was worth mentioning :).

Reply:
Hmm, I am not sure that within a transaction, you are unable to have a normal input and a stealth input being spent, just as you can have stealth outputs and non stealth outputs as shown in the pastebin I attached (line 89-101). In which case, we kind of have to go through each input.

Reply:
Moved the is_spendable check in the coin class. For any coin without stealth transactions, it's a noop defaulting to ```True```.

Also made a drive-by fix on latest master. In ```getrawtransactions()```, the second parameter is supposed to be ```int``` not ```bool``` (as is the case in ```getrawtransaction()``` where it gets converted from bool to int before being put on the wire)